### PR TITLE
pythonPackages.XlsxWriter: enable tests and pythonPackages.can: 3.0.0 -> 3.1.0

### PR DESCRIPTION
**Pull request by jluttine**
###### Motivation for this change

Update XlsxWriter and also fix+upgrade python-can because otherwise nox-review failed.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/can/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/can/default.nix]:
```suggestion
```
Those are no longer available in Nixpkgs.

File: pkgs/development/python-modules/can/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/can/default.nix]:
```suggestion
  propagatedBuildInputs = [ wrapt pyserial ] ++ lib.optional (pythonOlder "3.5") typing;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
The XlsxWriter bump will conflict with the staging-next branch. Let's wait until that is merged.

File: pkgs/development/python-modules/can/default.nix
Comment (jluttine) [reply pkgs/development/python-modules/can/default.nix]: Thanks, made the correction!

File: pkgs/development/python-modules/can/default.nix
Comment (jluttine) [reply pkgs/development/python-modules/can/default.nix]: Thanks, made the correction!

File: pkgs/development/python-modules/can/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/can/default.nix]: unused

Comment (jluttine) [reply pkgs/development/python-modules/can/default.nix]: Oh, yes. Thanks!

**Comment by jluttine** [comment]
Rebased on master.

**Comment by FRidh** [comment]
@GrahamcOfBorg build pythonPackages.XlsxWriter python3Packages.XlsxWriter pythonPackages.can python3Packages.can

**Comment by jluttine** [comment]
I accidentally first rebased on some really old state of master branch.. Now it should be fixed and I ran nox-review and everything was fine.

**Pull request merged**
